src/utils/api.tsx

```diff
@@ -274,6 +277,16 @@ export default class API {
 		}
 		const shifts = res.shifts.map((d: ShiftDto) => new Shift(d, course))
 
+		// Fetch the campus if not defined!
+		await Promise.all(shifts.map(async (shift) => {
+			const roomData = shift.getRoom()
+			if (!shift.isCampusDefined() && roomData) {
+				const spaceData = await this.getSpace(roomData.id)
+				const campusName = spaceData?.campus?.name
+				campusName && shift.setCampus(campusName)
```

Comment:
I'd rather have this setCampus be on a separate line with an if or something, don't really like using short circuiting for separate instructions